pkgs/development/libraries/gmp/6.x.nix

```
@@ -41,12 +44,12 @@ let self = stdenv.mkDerivation rec {
     # ARM optimization flags via /proc/cpuinfo (and is also
     # broken on multicore CPUs). Avoid this impurity.
     "--build=${stdenv.buildPlatform.config}"
-  ] ++ optional (cxx && stdenv.isDarwin) "CPPFLAGS=-fexceptions"
-    ++ optional (stdenv.isDarwin && stdenv.is64bit) "ABI=64"
+  ] ++ lib.optional (cxx && stdenv.isDarwin) "CPPFLAGS=-fexceptions"
+    ++ lib.optional (stdenv.isDarwin && stdenv.is64bit) "ABI=64"
     # to build a .dll on windows, we need --disable-static + --enable-shared
     # see https://gmplib.org/manual/Notes-for-Particular-Systems.html
-    ++ optional (!withStatic && stdenv.hostPlatform.isWindows) "--disable-static --enable-shared"
-    ++ optional (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64) "--disable-assembly";
+    ++ lib.optional (!withStatic && stdenv.hostPlatform.isWindows) "--disable-static --enable-shared"
```

Currently does not make a difference but it might be nice to make them a separate list items for once we switch to `_structuredAttrs`:

```suggestion
    ++ lib.optionals (!withStatic && stdenv.hostPlatform.isWindows) [
      "--disable-static"
      "--enable-shared"
    ]
```